x-pack/plugins/index_management/public/application/sections/enrich_policy_create/steps/fields/indices_selector.tsx

```diff
@@ -0,0 +1,116 @@
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useState, useCallback, useEffect } from 'react';
+import { i18n } from '@kbn/i18n';
+import { uniq } from 'lodash';
+import { EuiFormRow, EuiComboBox, EuiComboBoxOptionOption } from '@elastic/eui';
+import { getMatchingIndices } from '../../../../services/api';
+import type { FieldHook } from '../../../../../shared_imports';
+import { getFieldValidityAndErrorMessage } from '../../../../../shared_imports';
+
+interface IOption {
+  label: string;
+  options: Array<{ value: string; label: string; key?: string }>;
+}
+
+interface Props {
+  field: FieldHook;
+  [key: string]: any;
+}
+
+const getIndexOptions = async (patternString: string) => {
+  const options: IOption[] = [];
+
+  if (!patternString) {
+    return options;
+  }
+
+  const { data } = await getMatchingIndices(patternString);
+  const matchingIndices = data.indices;
+
+  if (matchingIndices.length) {
+    const matchingOptions = uniq([...matchingIndices]);
+
+    options.push({
+      label: i18n.translate('xpack.idxMgmt.enrichPolicyCreate.indicesSelector.optionsLabel', {
+        defaultMessage: 'Based on your indices',
+      }),
+      options: matchingOptions
+        .map((match) => {
+          return {
+            label: match,
+            value: match,
+          };
+        })
+        .sort((a, b) => String(a.label).localeCompare(b.label)),
+    });
+  } else {
+    options.push({
+      label: i18n.translate('xpack.idxMgmt.enrichPolicyCreate.indicesSelector.noMatchingOption', {
+        defaultMessage: 'No indices match your search criteria.',
+      }),
+      options: [],
+    });
+  }
+
+  return options;
+};
+
+export const IndicesSelector = ({ field, ...rest }: Props) => {
+  const { isInvalid, errorMessage } = getFieldValidityAndErrorMessage(field);
+  const [indexOptions, setIndexOptions] = useState<IOption[]>([]);
+  const [isIndiciesLoading, setIsIndiciesLoading] = useState<boolean>(false);
+
+  const onSearchChange = useCallback(
+    async (search: string) => {
+      setIsIndiciesLoading(true);
+      setIndexOptions(await getIndexOptions(search));
+      setIsIndiciesLoading(false);
+    },
+    [setIsIndiciesLoading, setIndexOptions]
+  );
+
+  // Load first 10 indices on mount so that the ComboBox has some options
+  useEffect(() => {
+    onSearchChange('*');
+  }, [onSearchChange]);
+
+  return (
+    <EuiFormRow
+      label={field.label}
+      labelAppend={field.labelAppend}
+      helpText={typeof field.helpText === 'function' ? field.helpText() : field.helpText}
+      error={errorMessage}
+      isInvalid={isInvalid}
+      fullWidth
+      {...rest}
+    >
+      <EuiComboBox
+        async
+        isLoading={isIndiciesLoading}
+        options={indexOptions}
+        noSuggestions={!indexOptions.length}
```

Comment:
This doesn't look like a critical issue to me, but decided to mention it, in case its solution doesn't require a lot of effort.

In the "Source indices" field, when we select an option, or remove all selected options, there is no dropdown menu with suggestions as usual (for example, the "Enrich fields" field always has this dropdown):

https://github.com/elastic/kibana/assets/59341489/22b10b25-89f4-43c8-a6da-9ed119567e84

The `indexOptions` list seems to be empty at these moments. I wonder if we could make it so that it always contains all index options.

Reply:
Nice catch! I've address it with https://github.com/elastic/kibana/pull/164131/commits/4698409198f51bd5e8f177cf1e771b7d8b752f31#diff-8d7aed67266fb3eb777d932177e99913f0a7d9c212e5a36bf49780c970f8cca8R80 so that we always keep the options visible even after clearing up the combobox